src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -45,89 +44,41 @@ public function indexAction(StructureInterface $structure, $preview = false, $pa
 
     /**
      * Creates a redirect for configured webspaces.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @deprecated since 1.2 use SuluWebsiteBundle:Redirect:redirectWebspace instead
      */
     public function redirectWebspaceAction(Request $request)
     {
-        $url = $this->resolveRedirectUrl(
-            $request->get('redirect'),
-            $request->getUri()
-        );
+        @trigger_error('SuluWebsiteBundle:Default:redirectWebspace is deprecated since version 1.2. Use the "SuluWebsiteBundle:Redirect:redirectWebspace" action instead.', E_USER_DEPRECATED);
 
-        return new RedirectResponse($url, 301);
+        return $this->forward(
```

Comment:
I would still prefer it. Don't you think that this can cause unexpected behavior, when thinking about caching and other stuff. Apart from that it is only a workaround to keep the deprecation working, so I don't think it is so bad, but at the same time I don't care about it too much.